synapse/lib/auth.py

```diff
@@ -375,11 +375,29 @@ async def feedBeholder(self, evnt, info, gateiden=None, logged=True):
 
             await self.fire('cell:beholder', **behold)
 
-    @s_nexus.Pusher.onPushAuto('user:info')
     async def setUserInfo(self, iden, name, valu, gateiden=None, logged=True, mesg=None):
 
         user = await self.reqUser(iden)
 
+        if name == 'locked' and not valu and user.isArchived():
+            raise s_exc.BadArg(mesg='Cannot unlock archived user.', user=iden, username=user.name)
+
+        if name in ('locked', 'archived') and not valu:
+            self.checkUserLimit()
```

Comment:
Do we need to keep the `self.checkUserLimit()` check in the bottom half of the handler?

Reply:
I had originally removed it from the bottom half but @invisig0th said to put it back so I did.